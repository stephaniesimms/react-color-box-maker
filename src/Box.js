import React, { Component } from 'react'

/** A single box in the box list
 * This has no state. Props background color, width, height, remove fn passed down from BoxList.
 * When each box component is displayed, add a button with text "X".
 * When user clicks button, remove the box.
 */

class Box extends Component {
  constructor(props) {
    super(props);
    this.handleRemove = this.handleRemove.bind(this);
  }

  handleRemove() {
    //call up the BoxList to remove box when user clicks button
    // console.log("handleRemove this.props", this.props)
    this.props.remove(this.props.id);
  }

  render() {
    const { backgroundColor, height, width } = this.props;

    return (
      <div>
        <div
          style={ {
            backgroundColor: backgroundColor,
            height: `${height}em`,
            width: `${width}em`
          } }
        />
        <button className="button"
          onClick={this.handleRemove}>X
        </button>
      </div>
    );
  }
}

export default Box;
