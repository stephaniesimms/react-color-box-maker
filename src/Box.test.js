import React from 'react';
import ReactDOM from 'react-dom';
import toJson from "enzyme-to-json";
import { mount, shallow } from 'enzyme';
import Box from './Box';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Box />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it("matches snapshot", function () {
  let wrapper = mount(<Box />);
  let serialized = toJson(wrapper);
  expect(serialized).toMatchSnapshot();
});

it("removes a box when button is clicked", function () {
  const removeFn = jest.fn();
  let wrapper = shallow(<Box remove={removeFn} />);
  wrapper.find("button").simulate("click");

  expect(removeFn).toHaveBeenCalled();
});