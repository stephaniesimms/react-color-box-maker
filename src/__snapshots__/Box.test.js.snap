// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<Box>
  <div>
    <div
      style={
        Object {
          "backgroundColor": undefined,
          "height": "undefinedem",
          "width": "undefinedem",
        }
      }
    />
    <button
      className="button"
      onClick={[Function]}
    >
      X
    </button>
  </div>
</Box>
`;
